### PR TITLE
docs: add CRUD subscriptions section

### PR DESCRIPTION
**Pull request by craicoverflow**
## Tasks

- [x] Added a section on Subscriptions to the CRUD documentation.

### CODE REVIEW

File: docs/crud/subscriptions.md [hunk 1]
```diff
@@ -4,4 +4,65 @@ title: Subscriptions
...
+  title: String!
+  likes: Int
+}
+
+// highlight-start
+input NoteSubscriptionFilter {
+  id: ID
+  title: String
+  likes: Int
+}
```
Comment (machi1990) [inline docs/crud/subscriptions.md:28]: We are finally keeping the filters?

File: docs/crud/subscriptions.md [hunk 2]
```diff
@@ -4,4 +4,65 @@ title: Subscriptions
...
 
-## TODO
\ No newline at end of file
+Subscriptions are divided into three groups of changes: **Create**, **Update** and **Delete**. 
+
+- `new<Type>`: subscribe to the [**Create**](./mutations#create) mutation event.
+- `updated<Type>`: subscribe to the [**Update**](./mutations#update) mutation event.
+- `deleted<Type>`: subscribe to the [**Delete**](./mutations#delete) mutation event.
+
+Graphback generates a subscription handler and filter input for each CRUD mutation field to let you subscribe to mutation events and receive real-time updates.
```
Comment (machi1990) [inline docs/crud/subscriptions.md:13]: Are we keeping the filters in this iteration?

File: packages/graphback-datasync/src/providers/DatasyncMongoDBDataProvider.ts [hunk 3]
```diff
@@ -135,6 +135,8 @@ export class DataSyncMongoDBDataProvider<Type = any> extends MongoDBDataProvider
         queryResult[fieldNames.updatedAt] !== undefined &&
         clientData[fieldNames.updatedAt].toString() !== queryResult[fieldNames.updatedAt].toString()
       ) {
+        queryResult[fieldNames.updatedAt] = queryResult[fieldNames.updatedAt].toString()
```
Comment (machi1990) [inline packages/graphback-datasync/src/providers/DatasyncMongoDBDataProvider.ts:138]: Not related. Need to rebase with master.

File: docs/crud/subscriptions.md (hunk 1 above)
Comment (craicoverflow) [reply docs/crud/subscriptions.md:28]: Woops! Need to take that bit out.

File: docs/crud/subscriptions.md [hunk 4]
```diff
@@ -4,4 +4,59 @@ title: Subscriptions
...
+
+Subscribing to a [`create`](./mutations#create) event on `Note`:
+
+```graphql
+newNote {
+  id
+  title
+  likes
+}
+```
```
Comment (machi1990) [inline docs/crud/subscriptions.md:42]:
```graphql
subscription {
newNote {
  id
  title
  likes
}
}
```
?

Likewise for update and delete

Comment (craicoverflow) [reply docs/crud/subscriptions.md]: Done now, for queries and mutations also.

File: docs/crud/subscriptions.md [hunk 5]
```diff
@@ -4,4 +4,75 @@ title: Subscriptions
...
+type Subscription {
+  newNote(filter: NoteSubscriptionFilter): Note!
+  updatedNote(filter: NoteSubscriptionFilter): Note!
+  deletedNote(filter: NoteSubscriptionFilter): Note!
+}
+// highlight-end
+```
+
+:::caution
+Subscription filtering is not available yet and will be implemented in a future release.
```
Comment (craicoverflow) [inline docs/crud/subscriptions.md:39]: Added this from https://github.com/aerogear/graphback/issues/1273#issuecomment-655489593

**Pull request merged**